### PR TITLE
Add more autotest sample

### PR DESCRIPTION
**Pull request by BBuf**
添加更多自动测试样例。
- [x] 修复一个自动测试框架打印代码段格式不正确的问题。
- [x] 按规范调整一些单测文件的位置。
- [x] bernoulli (OP存在BUG)
- [x] flow.mean & tensor.mean。
- [x] flow.sign & tensor.sign。 
- [x] flow.gather & tensor.gather，修复了一个dim和index顺序相反的bug和asser的bug。 
- [x] 添加了upsample几个测试样例，又发现Pytorch的两个BUG。暂时屏蔽了这个测试。
- [x]  给nn.Conv1d添加自动测试。
- [x] 给nn.ConvTranspose2d添加自动测试。
- [x] 给GroupNorm添加自动测试。

### CODE REVIEW

File: python/oneflow/nn/modules/normalization.py [hunk 1]
```diff
@@ -121,9 +121,9 @@ def forward(self, input: Tensor) -> Tensor:
             normalized, shape=[origin_shape[0], self.num_channels, -1]
         )
         if self.weight is not None:
-            normalized = normalized * self.weight
+            normalized = normalized * self.weight.reshape(1, self.num_channels, 1)
```
Comment (MARD1NO) [inline python/oneflow/nn/modules/normalization.py:124]: 为什么是放到后面进行reshape？

感觉改了跟没改一样

File: python/oneflow/test/modules/test_mean.py [hunk 2]
```diff
@@ -79,12 +79,12 @@ def test_mean(test_case):
...
-        arg_dict = OrderedDict()
-        arg_dict["test_type"] = [test_flow_against_pytorch, test_tensor_against_pytorch]
-        arg_dict["device"] = ["cpu", "cuda"]
-        for arg in GenArgList(arg_dict):
-            arg[0](test_case, "mean", device=arg[1])
+    @autotest()
+    def test_mean_random_data(test_case):
+        device = random_device()
+        dim = random(1, 4).to(int)
+        x = random_pytorch_tensor(ndim=4).to(device)
```
Comment (MARD1NO) [inline python/oneflow/test/modules/test_mean.py:86]: mean限制下数据类型，我印象里pytorch只支持float类型做mean，int会报错。虽然不影响autotest，但我感觉显示定义下会直观点

我那个align PR已经添加了这个检查

**Review by MARD1NO — APPROVED** [review]

File: python/oneflow/nn/modules/normalization.py (hunk 1 above)
Comment (BBuf) [reply python/oneflow/nn/modules/normalization.py:124]: 权重的shape需要和pytorch对齐

File: python/oneflow/test/modules/test_mean.py (hunk 2 above)
Comment (BBuf) [reply python/oneflow/test/modules/test_mean.py]: 已修改

File: oneflow/user/kernels/upsample_kernel.h [hunk 3]
```diff
@@ -27,9 +28,8 @@ OF_DEVICE_FUNC T GetLinearInputIndex(const int64_t out_dim_idx, const T scale, b
 
 OF_DEVICE_FUNC static int64_t GetNearestInputIndex(const int64_t out_dim_idx, const float scale,
                                                    const int64_t in_dim_size) {
-  int64_t index = static_cast<int64_t>(std::floor((static_cast<float>(out_dim_idx) * scale)));
+  int64_t index = static_cast<int64_t>(floorf(out_dim_idx * scale));
```
Comment (doombeaker) [inline oneflow/user/kernels/upsample_kernel.h:31]: 标准库的这个函数可以在 cuda 上运行的吗（因为这个函数上 `OF_DEVICE_FUNC`)？超出了我的理解。

File: python/oneflow/test/modules/test_bernoulli.py [hunk 4]
```diff
@@ -53,6 +54,12 @@ def test_bernoulli(test_case):
         for arg in GenArgList(arg_dict):
             arg[0](test_case, *arg[1:])
 
+    @unittest.skip("bernoulli has bug")
+    @autotest(auto_backward=False)
+    def test_flow_bernoulli_with_random_data(test_case):
+        input = random_pytorch_tensor(ndim=1).to("cpu")
+        return torch.bernoulli(input)
```
Comment (doombeaker) [inline python/oneflow/test/modules/test_bernoulli.py:61]: 这里我很好奇，这个算子上个随机函数，不固定种子（我不知道是否可以固定种子）的前提下，上几乎不可能保证 oneflow 和 PyTorch 的输出一致吧？

File: python/oneflow/test/modules/test_mean.py [hunk 5]
```diff
@@ -79,12 +79,12 @@ def test_mean(test_case):
...
             arg[0](test_case, *arg[1:])
 
-    def test_mean_against_pytorch(test_case):
-        arg_dict = OrderedDict()
-        arg_dict["test_type"] = [test_flow_against_pytorch, test_tensor_against_pytorch]
-        arg_dict["device"] = ["cpu", "cuda"]
-        for arg in GenArgList(arg_dict):
-            arg[0](test_case, "mean", device=arg[1])
+    @autotest()
+    def test_mean_random_data(test_case):
```
Comment (doombeaker) [inline python/oneflow/test/modules/test_mean.py:83]:
```suggestion
    def test_mean_with_random_data(test_case):
```

看其他名字带了 `with`

File: python/oneflow/test/modules/test_upsample.py [hunk 6]
```diff
@@ -396,6 +373,32 @@ def test_upsample2d(test_case):
         for arg in GenArgList(arg_dict):
             arg[0](test_case, *arg[1:])
 
+    @unittest.skip(
+        "pytorch nearest interpolate has bug, https://github.com/pytorch/pytorch/issues/65200"
```
Comment (doombeaker) [inline python/oneflow/test/modules/test_upsample.py:377]:
```suggestion
        "The nearest interpolate operation in pytorch has bug, https://github.com/pytorch/pytorch/issues/65200"
```
下同

File: python/oneflow/test/tensor/test_tensor.py [hunk 7]
```diff
@@ -581,12 +602,12 @@ def test_argsort_tensor_with_random_data(test_case):
...
         return y
 
-    @flow.unittest.skip_unless_1n1d()
-    def test_mean(test_case):
-        input = flow.tensor(np.random.randn(2, 3), dtype=flow.float32)
-        of_out = input.mean(dim=0)
-        np_out = np.mean(input.numpy(), axis=0)
-        test_case.assertTrue(np.allclose(of_out.numpy(), np_out, 0.0001, 0.0001))
+    @autotest()
+    def test_mean_random_data(test_case):
```
Comment (doombeaker) [inline python/oneflow/test/tensor/test_tensor.py:606]:
```suggestion
    def test_mean_tensor_random_data(test_case):
```

**Review by doombeaker — APPROVED** [review]

File: python/oneflow/test/modules/test_bernoulli.py (hunk 4 above)
Comment (BBuf) [reply python/oneflow/test/modules/test_bernoulli.py:61]: 这个测试跳过了，暂时没办法保证

File: python/oneflow/test/modules/test_mean.py (hunk 5 above)
Comment (BBuf) [reply python/oneflow/test/modules/test_mean.py:83]: 好的
